improver/lightning.py

```diff
@@ -204,3 +205,215 @@ def latitude_to_threshold(
+                    f"{cube.name()} and {apcp.name()} do not have the same spatial "
+                    f"coordinates"
+                )
+
+        return cape, liftidx, pwat, cin, apcp
+
+    def process(self, cubes: CubeList, model_id_attr: str = None) -> Cube:
+        """
+        From the supplied CAPE, Lifted Index, Precipitable Water, CIN, and 3-hr Accumulated
+        Precipitation cubes, calculate a probability of lightning cube.
+
+        Args:
+            cubes:
+                Cubes of CAPE, Lifted Index, Precipitable Water, CIN, and 3-hr Accumulated
+                Precipitation cubes.
+            model_id_attr:
+                The name of the dataset attribute to be used to identify the source
+                model when blending data from different models.
+
+        Returns:
+            Cube of lightning data
+
+        Raises:
+            ValueError:
+                If one of the cubes is not found or doesn't match the other
+        """
+        cape, liftidx, pwat, cin, apcp = self._get_inputs(cubes)
+
+        # Regression equations require math on cubes with incompatible units, so strip data
+        templ = apcp.copy()
+        cape = cape.data
+        liftidx = liftidx.data
+        pwat = pwat.data
+        cin = (
+            -1 * cin.data
+        )  # use inverse convention where CIN is positive instead of negative
+        apcp = apcp.data / 1000 * 39.3701  # convert kg m-2 to inches
+
+        # Regression equation when CAPE and APCP are greater than zero:
+        lprob = cape * apcp
+        lprob = 0.13 * np.log(lprob + 0.7) + 0.05
+
+        # If APCP is very low, a separate regression equation is used to predict lightning
+        # probability. The definition of “very low” is raised slightly when PWAT values are
+        # high. This is because the model often produces showery precipitation that doesn’t
+        # access the actual instability in very moist environments:
+        lprob_noprecip = cape / (cin + 100.0)
+        lprob_noprecip = 0.025 * np.log(lprob_noprecip + 0.31) + 0.03
+        apcp_temp = apcp - (pwat / 1000)
+
+        lprob[np.where(apcp_temp < 0.01)] = lprob_noprecip[np.where(apcp_temp < 0.01)]
+
+        # If there is no CAPE but the atmosphere is “close” to unstable, lightning does sometimes
+        # occur, especially when heavy precipitation may have stabilized the atmosphere in the
+        # model. Unstable values of lifted index are positive here:
+        liftidx = liftidx + 4.0
+
+        liftidx[liftidx < 0] = 0
+
+        lprob[np.where(cape <= 0)] = (
+            0.2 * (liftidx[np.where(cape <= 0)] * apcp[np.where(cape <= 0)]) ** 0.5
+        )
+
+        # Finally, the probability of lightning is reduced when there is not much PWAT, since
+        # graupel cannot form required for the charging process. The probability is reduced as:
+        lprob[np.where(pwat < 20)] = lprob[np.where(pwat < 20)] * (
+            pwat[np.where(pwat < 20)] / 20.0
+        )
```

Comment:
```suggestion
        low_pwat = np.where(pwat < 20)
        lprob[low_pwat] = lprob[low_pwat] * (pwat[low_pwat] / 20.0)
```
This change will give you a fractional speed up by reducing the number of tests.